include/picongpu/fields/FieldManipulator.kernel

```diff
@@ -19,93 +19,199 @@
+                    {
+                        cell[ axis ] += SuperCellSize::toRT()[ axis ] * -1 * relExchangeDir[ axis ];
+                        int factor(0);
+
+                        if( relExchangeDir[ axis ] < 0 )
+                        {
+                            factor = SuperCellSize::toRT()[ axis ] - cell[ axis ] +
+                                thickness - 1;
+                        }
+                        else
+                        {
+                            factor = SuperCellSize::toRT()[ axis ] + cell[ axis ] -
+                                localDomainCells[ axis ] + thickness;
+                        }
+
+                        if( factor <= 0 )
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            float_X const a = math::exp( -absorberStrength * float_X( factor ) );
+                            field( cell ) = field( cell ) * a;
+                        }
+                    } while( true );
+                }
+            );
 
         }
+    };
 
-        if (factor <= float_X(0.0))
-        {
-            factor = float_X(0.0);
-            nvidia::atomicAllExch(&finish,1);
-        }
-        else
+} // namespace detail
+
+    /** damp each field component at the one exchange border
+     *
+     * @tparam T_numWorkers number of workers
+     */
+    template< uint32_t T_numWorkers >
+    struct KernelAbsorbBorder
+    {
+        /** damp a field at the border
+         *
+         * @tparam T_BoxedMemory pmacc::DataBox, type of the field
+         * @tparam T_Mapping mapper functor type
+         *
+         * @param field filed to manipulate
+         * @param thickness the thickness of the absorber area (in cells)
+         * @param absorberStrength strength of the absorber
+         * @param mapper functor to map a block to a supercell
+         */
+        template<
+            typename T_BoxedMemory,
+            typename T_Mapping
+        >
+        DINLINE void operator()(
+            T_BoxedMemory field,
+            uint32_t thickness,
+            float_X absorberStrength,
+            T_Mapping mapper
+        ) const
         {
-            float_X a = math::exp(float_X(-absorber_strength) * (factor));
 
-            float3_X tmp = field(cell);
-            tmp.x() *= a;
-            tmp.y() *= a;
-            tmp.z() *= a;
-            field(cell) = tmp;
+            DataSpace< simDim > const relExchangeDir =
+                Mask::getRelativeDirections< simDim >( mapper.getExchangeType( ) );
+
+            /* create a sequence with int values [0;simDim)
+             * MakeSeq_t allows to use the result of mpl::range_c
+             * within the PMacc ForEach
+             */
+            using SimulationDimensions = MakeSeq_t<
+                boost::mpl::range_c<
+                    int,
+                    0,
+                    simDim
```

Comment:
Yes but we need the result as  `int` to compare it with a `DataSpace`. I will try to cast the value.